linera-client/src/client_context.rs

```diff
@@ -633,38 +631,8 @@ where
                 .expect("should execute block with OpenChain operations");
         }
         self.update_wallet_from_client(&chain_client).await?;
-        // Make sure all chains have registered the application now.
-        let futures = key_pairs
-            .keys()
-            .map(|&chain_id| {
-                let chain_client = self.make_chain_client(chain_id);
-                async move {
-                    for i in 0..5 {
-                        linera_base::time::timer::sleep(Duration::from_secs(i)).await;
-                        chain_client.process_inbox().await?;
-                        let chain_state = chain_client.chain_state_view().await?;
-                        if chain_state
-                            .execution_state
-                            .system
-                            .registry
-                            .known_applications
-                            .contains_key(&application_id)
-                            .await?
-                        {
-                            return Ok::<_, Error>(chain_client);
-                        }
-                    }
-                    panic!("Could not instantiate application on chain {chain_id:?}");
-                }
-            })
-            .collect::<Vec<_>>();
-        // We have to collect the futures to avoid a higher-ranked lifetime error:
-        // https://github.com/rust-lang/rust/issues/102211#issuecomment-1673201352
-        let clients = stream::iter(futures)
-            .buffer_unordered(max_in_flight)
-            .try_collect::<Vec<_>>()
-            .await?;
-        for client in clients {
+        for chain_id in key_pairs.keys() {
+            let client = self.make_chain_client(*chain_id);
             self.update_wallet_from_client(&client).await?;
         }
```

Comment:
This is a no-op now, isn't it? The whole `for` loop can go.

Reply:
My bad, yes